### PR TITLE
Support both old rotation field and new endpoints field

### PR DESCRIPTION
**Pull request by oyving**
Now we support both fields again in both serialization and deserialization to avoid issues with upgrades.

### CODE REVIEW

**Review by jonmv — APPROVED** [review]
I guess such a switch is a good use case for feature flags, but this should work, too :)
I think I prefer the two-step (write first, then delete) approach over the feature flags myself, tbh..

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java [hunk 1]
```diff
@@ -163,6 +163,7 @@ public Slime toSlime(Application application) {
         root.setDouble(queryQualityField, application.metrics().queryServiceQuality());
         root.setDouble(writeQualityField, application.metrics().writeServiceQuality());
         application.pemDeployKey().ifPresent(pemDeployKey -> root.setString(pemDeployKeyField, pemDeployKey));
+        application.legacyRotation().ifPresent(rotation -> root.setString(deprecatedRotationField, rotation.asString()));
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java]: The `legacyRotation` field isn't really needed as we could've just written `deprecatedRotationField` directly from `application.rotations()` here. This is fine for now though, but let's remove it in the next version.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java]: Yup. More important to have the issue fixed :) Removing it with the legacy (de)serialisation should be OK once this is deployed.

Comment (oyving) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java]: I agree, but I didn't feel brave enough to make assumptions about order or anything like that :)

**Pull request merged**